src/crypto/box_/curve25519xsalsa20poly1305.rs

```diff
@@ -89,6 +96,29 @@ pub fn seal(m: &[u8],
     c
 }
 
+/// `seal_detached()` encrypts and authenticates a message `m` using the senders secret key `sk`,
+/// the receivers public key `pk` and a nonce `n`. `m` is encrypted in place, so after this
+/// function returns it will contain the ciphertext. The detached authentication tag is returned by
+/// value.
+pub fn seal_detached(m: &mut [u8],
+                     &Nonce(ref n): &Nonce,
+                     &PublicKey(ref pk): &PublicKey,
+                     &SecretKey(ref sk): &SecretKey) -> Tag {
```

Comment:
I'm wondering if a better return type wouldn't be (&[u8], Tag), to improve the ergonomics of use. What do you think?